src/charm.py

```diff
@@ -261,38 +261,71 @@ def _get_gateway_apis_kubernetes_resource_manager(self):
             logger=LOGGER,
         )
 
-    def _workload_tracing_config(self) -> Dict[str, str]:
-        """Set workload tracing configuration."""
-        tracing_config = {}
-        if not self.workload_tracing.is_ready():
-            return tracing_config
-
-        workload_tracing_endpoint = self.workload_tracing.get_endpoint("otlp_grpc")
-        if not workload_tracing_endpoint:
-            return tracing_config
-
-        tracing_config["meshConfig.enableTracing"] = "true"
-        tracing_config["meshConfig.extensionProviders[0].name"] = "otel-tracing"
-        tracing_config["meshConfig.extensionProviders[0].opentelemetry.port"] = urlparse(
-            f"//{workload_tracing_endpoint}"
-        ).port
-        tracing_config["meshConfig.extensionProviders[0].opentelemetry.service"] = urlparse(
-            f"//{workload_tracing_endpoint}"
-        ).hostname
-        tracing_config["meshConfig.defaultProviders.tracing[0]"] = "otel-tracing"
-        tracing_config["meshConfig.defaultConfig.tracing.sampling"] = 100.0
-
-        return tracing_config
+    def _workload_tracing_provider(self) -> Tuple[List[Any], Dict[str, Any]]:
+        """Return a tuple with a list containing the tracing provider and global tracing settings."""
+        providers = []
+        global_config = {}
```

Comment:
nit: works fine like this, but since you only have one provider and one config to return, making the objs here and .appending them later is just extra code